src/model/test/CoilSystemCoolingWaterHeatExchangerAssisted_GTest.cpp

```diff
@@ -128,12 +133,23 @@ TEST_F(ModelFixture, CoilSystemCoolingWaterHeatExchangerAssisted_clone) {
   EXPECT_TRUE(coilSystem.inletModelObject());
   EXPECT_TRUE(coilSystem.outletModelObject());
 
-  EXPECT_FALSE(coilSystem2.airLoopHVAC());
-  EXPECT_FALSE(coilSystem2.inletModelObject());
-  EXPECT_FALSE(coilSystem2.outletModelObject());
+  EXPECT_FALSE(coilSystemClone.airLoopHVAC());
+  EXPECT_FALSE(coilSystemClone.inletModelObject());
+  EXPECT_FALSE(coilSystemClone.outletModelObject());
+
+  EXPECT_NE(coilSystemClone.coolingCoil().handle(), cc.handle());
+  EXPECT_NE(coilSystemClone.heatExchanger().handle(), hx.handle());
+
+  coilSystemClone.remove();
+  EXPECT_EQ(1, m.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+  EXPECT_EQ(1, m.getConcreteModelObjects<CoilCoolingWater>().size());
+
+  coilSystem.remove();
+  EXPECT_EQ(0, m.getConcreteModelObjects<CoilSystemCoolingWaterHeatExchangerAssisted>().size());
+  EXPECT_EQ(0, m.getConcreteModelObjects<CoilCoolingWater>().size());
 
-  EXPECT_NE(coilSystem2.coolingCoil().handle(), cc.handle());
-  EXPECT_NE(coilSystem2.heatExchanger().handle(), hx.handle());
+  EXPECT_EQ(2, a.supplyComponents().size());  // o --- o
+  EXPECT_EQ(5, p.demandComponents().size());  // o --- Splitter --- o --- Mixer --- o
```

Comment:
Extend testing for CoilSystemHXAssisted: they would fail before overriding remove(), the p.demandComponents was returning 0 because the connections were broken.